server/service/apple_mdm.go

```diff
@@ -2123,6 +2128,10 @@ func ReconcileProfiles(
 		resp := <-ch
 		if resp.Err != nil {
 			resp.Payload.CommandUUID = ""
+			// TODO(mna): so we set status to nil instead of pending so that it gets
+			// picked up again? This means that if we set it to pending immediately
+			// when a new profile is added or a host moves to a team, it would not
+			// get picked currently?
```

Comment:
with the current logic, as far as the cron is concerned, a profile with `pending` is either being removed or added, so it shouldn't be touched.

I _think_ It'll all depend on how we modify `ds.ListMDMAppleProfilesToInstall`, but I think you're right that with the current specs in https://github.com/fleetdm/fleet/issues/10122 this won't work.